### PR TITLE
Fix inspection frequency backend crash

### LINKED ISSUE #969: Incorrect InspectionFrequency crashes backend
**Describe the bug**
Sending a incorrect inspectionFrequency to the CustomMission API crashes the backend (SQL error, so will not work on SQLite db)
We probably need to sanitize the input for the inspectionFrequency.

**To Reproduce**
Create a custom mission, and add the following inspecetion frequency:
"inspectionFrequency": "5.00:0:0",

Exception log:

2023-08-11 14:25:59 - [41m[30mfail[39m[22m[49m: Microsoft.EntityFrameworkCore.Database.Command[20102]
      => SpanId:84290bae1a59bcc3, TraceId:8dd993524f2d4444445dec3acb695a50, ParentId:0000000000000000 => ConnectionId:0HMSQEQO99KUE => RequestPath:/missions/custom RequestId:0HMSQEQO99KUE:00000023 => Api.Controllers.MissionController.Create (api)
      Failed executing DbCommand (0ms) [Parameters=[@p0='?' (Size = 450), @p1='?' (Size = 4000), @p2='?' (DbType = Int32), @p3='?' (Size = 450), @p4='?' (Size = 450), @p5='?' (Size = 1000), @p6='?' (DbType = Time), @p7='?' (Size = 4000), @p8='?' (DbType = Boolean), @p9='?' (Size = 450), @p10='?' (Size = 200), @p11='?' (Size = 450)], CommandType='Text', CommandTimeout='30']
      SET NOCOUNT ON;
      INSERT INTO [Sources] ([Id], [SourceId], [Type])
      VALUES (@p0, @p1, @p2);
      INSERT INTO [MissionDefinitions] ([Id], [AreaId], [Comment], [InspectionFrequency], [InstallationCode], [IsDeprecated], [LastRunId], [Name], [SourceId])
      VALUES (@p3, @p4, @p5, @p6, @p7, @p8, @p9, @p10, @p11);
2023-08-11 14:25:59 - [41m[30mfail[39m[22m[49m: Microsoft.EntityFrameworkCore.Update[10000]
      => SpanId:84290bae1a59bcc3, TraceId:8dd993524f2d4444445dec3acb695a50, ParentId:0000000000000000 => ConnectionId:0HMSQEQO99KUE => RequestPath:/missions/custom RequestId:0HMSQEQO99KUE:00000023 => Api.Controllers.MissionController.Create (api)
      An exception occurred in the database while saving changes for context type 'Api.Database.Context.FlotillaDbContext'.
      Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while saving the entity changes. See the inner exception for details.
       ---> System.OverflowException: SqlDbType.Time overflow.  Value '5.00:00:00' is out of range.  Must be between 00:00:00.0000000 and 23:59:59.9999999.
         at Microsoft.Data.SqlClient.SqlCommand.<>c.<ExecuteDbDataReaderAsync>b__208_0(Task`1 result)
         at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
         at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
      --- End of stack trace from previous location ---
         at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
      --- End of stack trace from previous location ---
         at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
         at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
         at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
         --- End of inner exception stack trace ---
         at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
         at Microsoft.EntityFrameworkCore.SqlServer.Update.Internal.SqlServerModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
         at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
         at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
         at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
         at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
         at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(StateManager stateManager, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
         at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
         at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
      Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while saving the entity changes. See the inner exception for details.
       ---> System.OverflowException: SqlDbType.Time overflow.  Value '5.00:00:00' is out of range.  Must be between 00:00:00.0000000 and 23:59:59.9999999.
         at Microsoft.Data.SqlClient.SqlCommand.<>c.<ExecuteDbDataReaderAsync>b__208_0(Task`1 result)
         at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
         at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
      --- End of stack trace from previous location ---
         at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
      --- End of stack trace from previous location ---
         at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
         at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
         at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
         --- End of inner exception stack trace ---
         at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
         at Microsoft.EntityFrameworkCore.SqlServer.Update.Internal.SqlServerModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
         at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
         at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
         at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
         at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
         at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(StateManager stateManager, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
         at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
         at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
2023-08-11 14:25:59 - [41m[30mfail[39m[22m[49m: Microsoft.AspNetCore.Server.Kestrel[13]
      => SpanId:84290bae1a59bcc3, TraceId:8dd993524f2d4444445dec3acb695a50, ParentId:0000000000000000 => ConnectionId:0HMSQEQO99KUE => RequestPath:/missions/custom RequestId:0HMSQEQO99KUE:00000023
      Connection id "0HMSQEQO99KUE", Request id "0HMSQEQO99KUE:00000023": An unhandled exception was thrown by the application.
      Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while saving the entity changes. See the inner exception for details.
       ---> System.OverflowException: SqlDbType.Time overflow.  Value '5.00:00:00' is out of range.  Must be between 00:00:00.0000000 and 23:59:59.9999999.
         at Microsoft.Data.SqlClient.SqlCommand.<>c.<ExecuteDbDataReaderAsync>b__208_0(Task`1 result)
         at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
         at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
      --- End of stack trace from previous location ---
         at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
      --- End of stack trace from previous location ---
         at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
         at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
         at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
         --- End of inner exception stack trace ---
         at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
         at Microsoft.EntityFrameworkCore.SqlServer.Update.Internal.SqlServerModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
         at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
         at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
         at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
         at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
         at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(StateManager stateManager, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
         at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
         at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
         at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
         at Api.Services.MissionDefinitionService.Create(MissionDefinition missionDefinition) in /source/api/Services/MissionDefinitionService.cs:line 53
         at Api.Controllers.MissionController.Create(CustomMissionQuery customMissionQuery) in /source/api/Controllers/MissionController.cs:line 502
         at lambda_method4836(Closure , Object )
         at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.AwaitableObjectResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
         at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Logged|12_1(ControllerActionInvoker invoker)
         at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
         at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
         at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
         at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
         at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|20_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
         at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Logged|17_1(ResourceInvoker invoker)
         at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Logged|17_1(ResourceInvoker invoker)
         at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
         at Microsoft.AspNetCore.Authorization.Policy.AuthorizationMiddlewareResultHandler.HandleAsync(RequestDelegate next, HttpContext context, AuthorizationPolicy policy, PolicyAuthorizationResult authorizeResult)
         at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
         at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
         at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
         at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
         at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application)

### PR DESCRIPTION
**Pull request by andchiind**
Closes #969

### CODE REVIEW

**Comment by andchiind** [comment]
It doesn't let me change InspectionFrequency directly to bigint from Time format, but since the inspection frequency column is empty anyways in each environment it is not an issue to jsut delete and readd the column.

**Comment by andchiind** [comment]
It's worth noting that these database changes have already been applied to dev

File: backend/api/appsettings.Development.json [hunk 1]
```diff
@@ -41,6 +41,6 @@
     "AccountName": "flotilladevsa"
   },
   "Database": {
-    "UseInMemoryDatabase": true
+    "UseInMemoryDatabase": false
```
Comment (OleDrange) [inline backend/api/appsettings.Development.json:44]: For my curiosity, why is this needed/ what is the benefit of this?  "UseInMemoryDatabase": false

Comment (Afonso-2403) [reply backend/api/appsettings.Development.json:44]: I also have the same question, wouldn't it make sense to keep it as true given that these are the settings that are loaded when we run flotilla locally?
If we set it to false we will be polluting the dev database when doing local testing, correct?

File: backend/api/Migrations/20230830075517_RemoveInspectionFrequency.cs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        {
+            migrationBuilder.DropColumn(
+                name: "InspectionFrequency",
+                table: "MissionDefinitions");
+
+            migrationBuilder.AlterColumn<string>(
+                name: "InstallationId",
+                table: "Plants",
+                type: "nvarchar(450)",
+                nullable: false,
```
Comment (Afonso-2403) [inline backend/api/Migrations/20230830075517_RemoveInspectionFrequency.cs:22]: Why are these changes being done if they are already in migration [RequirePlantAndInstallation](https://github.com/equinor/flotilla/blob/main/backend/api/Migrations/20230811072130_RequirePlantAndInstallation.cs)?

**Review by Afonso-2403 — CHANGES_REQUESTED** [review]
The migrations should probably be regenerated so that they don't include making a bunch of properties required when this PR is only about changing the type of the Inspection frequency

File: backend/api/appsettings.Development.json (hunk 1 above)
Comment (oysand) [reply backend/api/appsettings.Development.json:44]: I think this should probably still stay as true

File: backend/api/Migrations/20230830075517_RemoveInspectionFrequency.cs (hunk 2 above)
Comment (andchiind) [reply backend/api/Migrations/20230830075517_RemoveInspectionFrequency.cs:22]: These are automatically generated so the underlying script must have detected that the change was not correctly in the model snaphsot until now. It is generally not a good idea to make manual changes here as this could mess up future auto-generated migrations, and it is difficult to correctly maintain the up/down functions as well as the snapshot.

Comment (andchiind) [reply backend/api/Migrations/20230830075517_RemoveInspectionFrequency.cs:22]: I will try to regenerate them though to see if that fixes it

Comment (andchiind) [reply backend/api/Migrations/20230830075517_RemoveInspectionFrequency.cs]: I can't see how removing this redundant code will break anything, so I just manually simplified the migration.

**Review by Afonso-2403 — APPROVED** [review]
LGTM
